### PR TITLE
PI-2306 populate licence conditions for cases with multiple custodial events

### PR DESCRIPTION
**Pull request by achimber-moj**

### CODE REVIEW

File: projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt [hunk 1]
```diff
@@ -65,13 +65,42 @@ class LicenceConditionApplier(
...
+                ),
+                activatedLicence,
+                occurredAt
             )
         }
+
+        //calculated end date should always be populated
+        sentences
+            .maxBy { it.disposal.endDate }.let {
+                return applyLicenceConditions(
```
Comment (marcus-bcl) [inline projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt:98]: Is this potentially adding the licence conditions to two of the events?  i.e. the one with the latest `enteredSenetenceEndDate`, and then the one with the latest `endDate`?

File: projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt [hunk 2]
```diff
@@ -65,13 +65,42 @@ class LicenceConditionApplier(
...
+    }
+
+    fun associateLicenceToLongestSentence(
+        sentences: List<Custody>,
+        com: PersonManager,
+        activatedLicence: ActivatedLicence,
+        occurredAt: ZonedDateTime,
+        properties: Map<String, String>
+    ): List<ActionResult> {
+        val longestSentenceByEnteredDate = sentences
```
Comment (marcus-bcl) [inline projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt:79]: Does longest mean the latest by end date here?  Or are we assuming the start date is the same on all matching sentences?

Comment (achimber-moj) [reply projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt:79]: i mean latest by end date.  i will rename the method to make it clearer.

File: projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt (hunk 1 above)
Comment (achimber-moj) [reply projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/LicenceConditionApplier.kt:98]: it should not to, as line 84 should return if the condition is met.

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**